salt/_modules/metalk8s_package_manager_yum.py

```diff
@@ -143,3 +152,62 @@ def list_pkg_dependents(
             del all_pkgs[pkg_name]
 
     return all_pkgs
+
+
+def _disable_yum_loggers():
+    yum_loggers = [
+        'yum.filelogging.RPMInstallCallback', 'yum.verbose.Repos',
+        'yum.verbose.plugin', 'yum.Depsolve', 'yum.verbose', 'yum.plugin',
+        'yum.Repos', 'yum', 'yum.verbose.YumBase', 'yum.filelogging',
+        'yum.verbose.YumPlugins', 'yum.RepoStorage', 'yum.YumBase',
+        'yum.filelogging.YumBase', 'yum.verbose.Depsolve',
+    ]
+
+    for logger_name in yum_loggers:
+        yum_logger = logging.getLogger(logger_name)
+        yum_logger.setLevel(yum.logginglevels.__NO_LOGGING)
```

Comment:
Why not just do:

```python
yum_logger = logging.getLogger('yum')
yum_logger.setLevel(yum.logginglevels.__NO_LOGGING)
```

?

Reply:
Before it's not sufficient. :)

Reply:
Oh, I get why: it's not the level that we want to change, we want to suppress messages from all child loggers of `yum`. Since the logger search is made top-down, if a higher-level logger (say `yum.verbose`) is set to a sufficiently low level, its handlers (and parent handlers) will be used. 

So, if the developers chose to explicitly set levels for their loggers, changing the level of the parents won't prevent the `LogRecord` from being created.

It is also possible to adjust the level on a `handler`, but since we don't want to manage this one (Salt does it), it's not an option.
What we could do, to filter all those messages using the parent logger, is use a `Filter`:

```python
import logging

class Suppress:
    def filter(self, record):
        return False

yum_logger = logging.getLogger('yum')
yum_logger.addFilter(Suppress())
```

Reply:
Still not sufficient. :)
```
# ./test.py
Package kubectl-1.15.3-0.x86_64 already installed and latest version
Package cri-tools-1.13.0-0.x86_64 already installed and latest version
Package metalk8s-sosreport-2.4-1.el7.x86_64 already installed and latest version
Package salt-minion-2018.3.4-1.el7.noarch already installed and latest version
Package containerd-1.2.4-1.el7.x86_64 already installed and latest version
Package calico-cni-plugin-3.8.2-1.el7.x86_64 already installed and latest version
Package kubelet-1.15.3-0.x86_64 already installed and latest version
Package m2crypto-0.31.0-3.el7.x86_64 already installed and latest version
Package kubernetes-cni-0.7.5-0.x86_64 already installed and latest version
```

Reply:
Oh, I checked out `yum`'s code, and it seems they actually use `propagate=False` for a few loggers, like `yum` and `yum.verbose`, but also `yum.filelogging` used for syslog apparently: https://github.com/rpm-software-management/yum/blob/master/yum/logginglevels.py#L150-L160

IMO, the cleanest would thus be to disable those three specifically, either with a filter, or using the internal `yum.logginglevels.__NO_LOGGING` as you did. Tested on `docker run -it --rm centos:7` :)